### PR TITLE
[core] Make `karma` run in parallel

### PR DESCRIPTION
**Pull request by romgrk**
Enable the `karma-parallel` plugin to run the test suite in parallel, which speeds it up quite a bit locally. Not sure if the CI has enough CPU cores or memory to be worth it, but for local runs it's much more efficient.

### CODE REVIEW

File: test/karma.conf.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ module.exports = function setKarmaConfig(config) {
         timeout: (process.env.CIRCLECI === 'true' ? 5 : 2) * 1000,
       },
     },
-    frameworks: ['mocha', 'webpack'],
+    frameworks: ['parallel', 'mocha', 'webpack'],
```
Comment (MBilalShafi) [inline test/karma.conf.js:23]: Seems like it's getting [out of resources](https://app.circleci.com/pipelines/github/mui/mui-x/50037/workflows/9cc2ed86-d2bc-4dad-b2ca-0ac7c45e519e/jobs/286969), probably adding a concurrency [limit](https://github.com/joeljeske/karma-parallel#options) (like 8) will do, should still speedup by up to 8x, wdyt?

**Comment by oliviertassinari** [comment]
To see how this behaves in https://automate.browserstack.com/dashboard/v2/builds/.

**Review by cherniavskii — APPROVED** [review]

**Comment by LukasTy** [comment]
Thanks for a nice improvement! 👍 
Did you or anyone else from the `xGrid` team investigate the single failing test in CI?
Could it have been a problem with the test itself? 🤔 
https://app.circleci.com/pipelines/github/mui/mui-x/50168/workflows/d4c4f5d3-c5b7-4aaa-987f-190c7e35ca0a/jobs/287658

**Comment by romgrk** [comment]
I did not. Looked like a focus issue, not sure if the parallelization in a virtual rendering context might be causing bugs with focus management. Didn't have enough bandwidth to investigate.

**Pull request merged**